osu.Game/Rulesets/Mods/ModFlashlight.cs

```diff
@@ -27,21 +28,47 @@ namespace osu.Game.Rulesets.Mods
 {
     public abstract class ModFlashlight : Mod
     {
+        private const int min_change_size_combo = 1;
+        private const int max_change_size_combo = 300;
+
         public override string Name => "Flashlight";
         public override string Acronym => "FL";
         public override IconUsage? Icon => OsuIcon.ModFlashlight;
         public override ModType Type => ModType.DifficultyIncrease;
         public override LocalisableString Description => "Restricted view area.";
 
-        [SettingSource("Flashlight size", "Multiplier applied to the default flashlight size.")]
-        public abstract BindableFloat SizeMultiplier { get; }
+        protected ModFlashlight()
+        {
+            FinalFlashlightSize.DefaultChanged += _ => FinalFlashlightSize.SetDefault();
+        }
 
-        [SettingSource("Change size based on combo", "Decrease the flashlight size as combo increases.")]
-        public abstract BindableBool ComboBasedSize { get; }
+        [SettingSource("Starting flashlight size", "Multiplier applied to the default flashlight size.")]
+        public abstract BindableFloat StartingFlashlightSize { get; }
+
+        [SettingSource("Final flashlight size", "Multiplier applied to the starting flashlight size after the max flashlight combo is reached.")]
+        public BindableFloat FinalFlashlightSize { get; } = new BindableFloat(0.8f)
+        {
+            MinValue = 0.5f,
+            MaxValue = 1,
+            Precision = 0.1f,
+        };
+
+        [SettingSource("Change size combo", "Changes the combo multiplier where the flashlight size is changed.")]
+        public BindableInt ChangeSizeCombo { get; } = new BindableInt(100)
+        {
+            MinValue = min_change_size_combo,
+            MaxValue = max_change_size_combo,
+        };
+
+        [SettingSource("Amount of size changes", "The amount of times the flashlight size changes.")]
```

Comment:
So this is a setting from 1 to 300...? That sounds like an abhorrent UX.

Reply:
The reason for it to being from 1 to 300 is so if a user wants they can just set the amount of size changes to any smaller value and set a custom combo multiplier.
e.g if you want the fl area to scale linearly until you reach 300 combo you set the amount of size changes to 300 and the combo multiplier to 1.

Reply:
Yes, I understand that. But most users will not. It requires serious brainpower to understand what this setting means (and how it affects gameplay as a divisor of intervals).

Reply:
Maybe a `FinalChangeSizeCombo` will do then? it seems more understandable to the average user atleast, it would be a multiple of the `ChangeSizeCombo`.

Reply:
> Yes, I understand that. But most users will not. It requires serious brainpower to understand what this setting means (and how it affects gameplay as a divisor of intervals).


Changed this in latest commit, so it is based on the `FinalChangeSizeCombo` instead.